Infrastructure/Data/Configurations/ProductConfiguration.cs

```diff
@@ -1,22 +1,27 @@
 using Domain.Models;
+using Infrastructure.Options;
 using Microsoft.EntityFrameworkCore.Metadata.Builders;
 
 namespace Infrastructure.Data.Configurations;
 
-public class ProductConfiguration : BaseEntityConfiguration<Product>
+public class ProductConfiguration(ProductValidationOptions productValidationOptions) : BaseEntityConfiguration<Product>
 {
     public override void Configure(EntityTypeBuilder<Product> builder)
     {
         base.Configure(builder);
         builder.Property(p => p.Name)
+           .HasMaxLength(productValidationOptions.NameMaxLength)
             .IsRequired();
         builder.Property(p => p.Description)
+           .HasMaxLength(productValidationOptions.DescriptionMaxLength)
             .IsRequired();
         builder.Property(p => p.AvailableAmount)
             .IsRequired();
         builder.Property(p => p.TotalAmountSold)
             .IsRequired();
         builder.Property(p => p.Rating)
+           .HasMaxLength(productValidationOptions.RatingMaxLength)
+           .HasMaxLength(productValidationOptions.RatingMinLength)
```

Comment:
Delete one with Min Length, because it makes NO sense. There couldnt be any MinLength configuration , unfortunately, so we gonna leave it this way(just with max length)